src/common/xo/migrate-vm-modal/index.js

```diff
@@ -163,6 +164,33 @@ export default class MigrateVmModalBody extends BaseComponent {
 
     // Intra-pool
     const defaultSr = pools[host.$pool].default_SR
+
+    let targetSr
+    if (
+      isSrShared(this._getObject(defaultSr)) ||
+      some(host.$PBDs, pbd => this._getObject(pbd).SR === defaultSr)
+    ) {
+      targetSr = defaultSr
+    } else {
+      for (let pbd of host.$PBDs) {
+        const sr = this._getObject(this._getObject(pbd).SR)
+        if (isSrShared(sr) && isSrWritable(sr)) {
+          targetSr = sr.id
+          break
+        }
+      }
+
+      if (targetSr === undefined) {
+        for (let pbd of host.$PBDs) {
+          const sr = this._getObject(this._getObject(pbd).SR)
+          if (isSrWritable(sr)) {
+            targetSr = sr.id
+            break
+          }
+        }
+      }
```

Comment:
```js
let targetSr
forEach(this.props.srsByHost[host.id], sr => {
  if (isSrWritable(sr)) {
    targetSr = sr.id
    if (isSrShared(sr)) {
      return false
    }
  }
})
```
(returning `false` exits the iteration of the [`forEach`](http://devdocs.io/lodash~4/index#forEach))

What do you think?